diffkemp/building/cc_wrapper/cc_wrapper_utils.h

```diff
@@ -0,0 +1,32 @@
+#ifndef CC_WRAPPER_UTILS_H
+#define CC_WRAPPER_UTILS_H
+
+#include <algorithm>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+#include <vector>
+
+// Utility functions for the cc_wrapper
+template <typename Container> Container split(const std::string &s, char delim);
+bool endsWith(const std::string &str, const std::string &suffix) noexcept;
+std::vector<std::string> getClangDefaultOptions(bool defaultOptim = false);
+bool isExecutable(const std::string &path) noexcept;
+bool isInVector(const std::string &s,
+                const std::vector<std::string> &vector) noexcept;
+struct WrapperArgs {
+    std::string dbFilename;
+    std::string clang;
+    std::vector<std::string> clangAppend;
+    std::unordered_set<std::string> clangDrop;
+    bool debug = false;
+    std::string llvmLink;
+    std::string llvmDis;
+    bool noOptOverride = false;
+    std::vector<std::string> clangArgs;
```

Comment:
These are the arguments of the original (typically GCC) command, right? And they are not just use for Clang but also for the GCC call so the name is not the best. How about we call it `cc_args` or (respecting Makefile conventions) `cflags` or `orig_cflags`?